### DISCUSSION TITLE
websockets.py send_json() doesnt' allow custom encoder

### DISCUSSION THREAD

**Discussion by amalizzio** [encode/starlette#2115] [Potential Issue] [answered]
I recently switched from django to fastapi and django allows in most places to pass custom encoder .

I have datetimes in objects and nan's which need special treatment.

My workaround is json.dumps(json.loads(deep_dictionary,cls=MyEncoder)) which is not efficient.

Should you pass through in send_json a custom encoder to the json.dumps please?

Thank you,

Anthony

**Comment by Kludex** [comment, marked as answer]
For the time being, the way to do it is to implement the logic of `send_json` yourself i.e. call `WebSocket.send()`.
